src/player.c

```diff
@@ -267,6 +267,37 @@ mpd_run_random(struct mpd_connection *connection, bool mode)
 		mpd_response_finish(connection);
 }
 
+bool
+mpd_send_single_state(struct mpd_connection *connection,
+		      enum mpd_single_state state)
+{
+	char state_str[16];
```

Comment:
I think this buffer is rather pointless overhead. Can you refactor this to `const char *state_str = single_state_to_string(state)`, which returns string literals?

Reply:
Sure, no problem. I've added a new commit

Reply:
Thanks, looks very good!